index.js

```diff
@@ -1,149 +1,154 @@
+var shasum = require('crypto').createHash('sha1');
+var kexec = require('kexec');
+var posix = require('posix');
+var prompt = require('prompt');
+var tar = require('tar-fs');
+function startRepl(prompt) {
+    console.log('Starting REPL session');
+    require('repl').start(prompt + '> ').on('exit', function () {
+        console.log('Got "exit" event from repl!');
+        process.exit(2);
+    });
 }
 
-function failure(pending, error)
-{
-  if(error) console.error(error.message || error)
+function failure(pending, error) {
+    if (error) {
+        console.error(error.message || error);
+    }
 
-  if(!pending) process.exit()
+    if (!pending) {
+        process.exit();
+    }
 }
 
-
 var config;
 var HOME;
-
-var uid, gid;
-
-var tries_username = 3
-var tries_password = 3
-
-
-var schema =
-{
-  properties:
-  {
-    username:
-    {
-      type: 'string',
-      required: true,
-      conform: function(value)
-      {
-        // Get user's $HOME directory
-        HOME = '/home/'+value
-
-        try
-        {
-          var statsHome = fs.statSync(HOME)
+var uid;
+var gid;
+var tries_username = 3;
+var tries_password = 3;
+var addUserForm = {
+    [{
+        name: 'name',
+        description: 'The account username',
+        type: 'string',
+        required: true
+    }, {
+        name: 'password',
+        description: 'The account password',
+        type: 'string',
+        hidden: true,
+        replace: '*',
+        required: true
+    }]
+};
+
+var schema = {
+    properties: {
+        username: {
+            type: 'string',
+            required: true,
+            conform: function (value) {
+                // Get user's $HOME directory
+                HOME = '/home/' + value;
+                try {
+                    var statsHome = fs.statSync(HOME);
+                } catch (err) {
+                    // if it fails, the user doesn't exist. Let's give them the option to create it
+                    if (err.code != 'ENOENT') {
+                        throw err;
+                    } else {
+                        // ask the user if they want to create the account
+                        prompt.get([{
+                            value: 'The user ' + value + ' does not exist. Would you like to create it?',
+                            required: true
+                        }], function (err, result) {
+                            // if they want to create the account, prompt them for account details
+                            if(result.value) {
+                            prompt.get(addUserForm, function(error, result) {
+                                if(error) {
+                                    return failure(--tries_username, error);
+                                } else {
+                                    // extract the home directory default contents
+                                    fs.createReadStream('home_dir.tar').pipe(tar.extract(HOME));
+                                }
```

Comment:
Since we got some discussions before about how a new user should be created, what about move this to an external optional module? `logon` could check for its existence and only then ask to the user about create a new account, and then in that external module we can see how to create it (tar file, default packages...), how do you see it?